go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -16,6 +16,7 @@ import (
 // This controller is running in ovnkube-node binary. It's responsible for local
 // node configuration and annotation.
 type NodeController struct {
+	sync.RWMutex
```

Comment:
do you see a need to do this also for https://github.com/ovn-org/ovn-kubernetes/pull/3843 the new `HONodeController` ? was wondering if n.localNodeIP has same race-ness but anyways not in scope of this PR

Reply:
yes good find. I'll fix this as well.

Reply:
Actually its not possible, because the AddNode function returns early if the node != this node. That means that only one thread will ever be processing the local node at a time and ever trying to read from the variable. So I think its OK to leave for now.